analysis/thread-dump/src/main/java/org/eclipse/jifa/tda/ThreadDumpAnalyzer.java

```diff
@@ -305,4 +297,31 @@ public Map<MonitorState, Integer> threadCountsByMonitor(int id) {
         map.forEach((s, l) -> counts.put(s, l.size()));
         return counts;
     }
+
+    /**
+     * @return rows of a thread dump table
+     */
+     public List<ThreadDumpRow> rows(PagingRequest paging) {
+
+         List<ThreadDumpRow> threadDumpRows = new ArrayList<>(snapshot.getJavaThreads().stream().map(t -> {
```

Comment:
style:

We can add static functions and/or local functions somewhere to do this conversion. For example, a `ThreadDumpRow(JavaThread javaThread)` constructor or a `static ThreadDumpRow fromJavaThread(JavaThread javaThread)`.

Though, it's a bit clunky having that in the Model, so maybe somewhere here.